integrations/oci/objectstorage/src/main/java/io/helidon/integrations/oci/objectstorage/OciObjectStorageHealthCheck.java

```diff
@@ -0,0 +1,233 @@
+     *
+     * @return builder instance
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Create an instance.
+     *
+     * @return an instance.
+     */
+    public static OciObjectStorageHealthCheck create() {
+        return builder().build();
+    }
+
+    /**
+     * Checks that the OCI Object Storage bucket is accessible, if defined. Will report
+     * error only if the bucket has been defined and it is not accessible for some reason.
+     * Can block since all health checks are called asynchronously.
+     *
+     * @return a response
+     */
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("objectStorage");
+        if (bucket != null) {
+            // Attempt to retrieve bucket's metadata
+            Single<ApiOptionalResponse<GetObjectRx.Response>> single =
+                    ociObjectStorage.getBucket(GetObject.Request.builder().bucket(bucket));
+            try {
+                ApiOptionalResponse<GetObjectRx.Response> r = single.get(timeout, TimeUnit.SECONDS);
+                builder.state(r.status().equals(OK_200));
+                LOGGER.fine(() -> "OCI ObjectStorage health check for bucket " + bucket
+                        + " returned status code " + r.status().code());
+            } catch (Throwable t) {
+                builder.state(false);
+            } finally {
+                builder.withData("bucket", bucket);
+            }
+        } else {
+            LOGGER.fine("OCI ObjectStorage health check disabled when bucket not defined");
+            builder.state(true);
+        }
+        builder.withData("namespace", namespace);
+        return builder.build();
+    }
+
+    /**
+     * Fluent API builder for {@link OciObjectStorageHealthCheck}.
+     */
+    public static final class Builder implements io.helidon.common.Builder<OciObjectStorageHealthCheck> {
+
+        private Config config;
+        private String bucket;
+        private String namespace;
+        private int timeout = -1;
+        private OciObjectStorageRx ociObjectStorage;
+
+        private Builder() {
+        }
+
+        @Override
+        public OciObjectStorageHealthCheck build() {
+            return new OciObjectStorageHealthCheck(this);
+        }
+
+        /**
+         * Set up this builder using config.
+         *
+         * @param config the config.
+         * @return the builder.
+         */
+        public Builder config(Config config) {
+            this.config = config;
+            Config timeout = config.get("objectstorage.healthcheck.timeout");
+            timeout.asInt().ifPresent(this::timeout);
+            return this;
+        }
+
+        /**
+         * Set timeout in millis.
+         *
+         * @param timeout timeout in millis.
+         * @return the builder.
+         */
+        public Builder timeout(int timeout) {
```

Comment:
Public builder method should be
```
public Builder timeout(Duration timeout) {
            this.timeout = timeout;
            return this;
}
```
A helper private method can be created that accepts configured value, such as `timeoutMillis(long)` or `timeoutSeconds(long)` that will call this method:
```
private timeoutSeconds(long seconds) {
  timeout(Duration.ofSeconds(seconds));
}
```